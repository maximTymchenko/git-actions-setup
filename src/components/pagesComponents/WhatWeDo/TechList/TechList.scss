@import '../../../../styles/vars';

.tech-list {
    &__container {
        display: flex;
        margin-bottom: 70px;
        justify-content: space-between;
    }

    &__wrapper {
        width: 510px;

        &--right {
            margin-right: 50px;
            width: 350px;
            border-right: 2px solid $section__border;
        }

        &--bottom {
            border-bottom: 2px solid $section__border;
        }

        @media screen and (max-width: $tablet) {
            border-right: none;
            margin-bottom: 40px;

            border-bottom: 2px solid $section__border;
        }

        @media screen and (max-width: $mobile) {
            width: 350px;
            margin: 0;
            margin-bottom: 20px;

            &:nth-child(even) {
                margin-top: 50px;
            }
        }
    }

    &__title {
        font-family: 'OktaNeue-Black';
        font-size: 28px;
        font-weight: 900;
        color: $text;
        position: relative;
        padding-bottom: 20px;

        &--hide {
            display: none;
        }

        // &::before {
        //     content: url('../../../../assets/img/small-waves.svg');
        //     position: absolute;
        //     left: -20px;
        //     top: -10px;
        //     z-index: -20;

        //     @media screen and (max-width: $mobile) {
        //         left: 5px;
        //     }
        // }
    }

    &__item {
        font-family: 'OktaNeue-Black';
        font-size: 15px;
        font-weight: 900;
        letter-spacing: 0.5px;
        color: $primary__light;
        margin-bottom: 15px;
        cursor: pointer;
    }
}
