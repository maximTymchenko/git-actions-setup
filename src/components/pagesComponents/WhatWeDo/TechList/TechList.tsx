import React, { FC } from 'react';
import './TechList.scss';

interface ITechListProps {
    title: string;
    techList: string[];
    hide: string;
    borderType: string;
}

const TechList: FC = (props: ITechListProps) => {
    return (
        <div
            data-testid="tech-list"
            className={`tech-list__wrapper tech-list__wrapper--${props.borderType}`}
        >
            <h3 className={`tech-list__title tech-list__title--${props.hide}`}>{props.title}</h3>
            <ul className="tech-list__items">
                {props.techList.map((item, index) => (
                    <li key={index} className="tech-list__item">
                        {item}
                    </li>
                ))}
            </ul>
        </div>
    );
};

export default TechList;
