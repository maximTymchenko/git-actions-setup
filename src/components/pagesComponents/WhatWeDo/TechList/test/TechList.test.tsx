import React from 'react'
import TechList from '../TechList'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
const webDevelopmenList = [
  'Architecture planning',
  'Microservice architecture',
  'Event sourcing',
  'CQRS',
  'Amazon AWS',
  'Google Firebase',
  'Microsoft Azure',
]

it('renders elem correctly', () => {
  const { getByTestId } = render(
    <TechList techList={webDevelopmenList} title="Web development" />
  )
  expect(getByTestId('tech-list')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<TechList techList={webDevelopmenList} title="Web development" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
