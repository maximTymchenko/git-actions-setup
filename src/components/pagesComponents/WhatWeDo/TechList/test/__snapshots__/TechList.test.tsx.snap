// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="tech-list__wrapper tech-list__wrapper--undefined"
  data-testid="tech-list"
>
  <h3
    className="tech-list__title tech-list__title--undefined"
  >
    Web development
  </h3>
  <ul
    className="tech-list__items"
  >
    <li
      className="tech-list__item"
    >
      Architecture planning
    </li>
    <li
      className="tech-list__item"
    >
      Microservice architecture
    </li>
    <li
      className="tech-list__item"
    >
      Event sourcing
    </li>
    <li
      className="tech-list__item"
    >
      CQRS
    </li>
    <li
      className="tech-list__item"
    >
      Amazon AWS
    </li>
    <li
      className="tech-list__item"
    >
      Google Firebase
    </li>
    <li
      className="tech-list__item"
    >
      Microsoft Azure
    </li>
  </ul>
</div>
`;
